lib/DefinePlugin.js

```diff
@@ -308,8 +309,10 @@ const PLUGIN_NAME = "DefinePlugin";
 const VALUE_DEP_PREFIX = `webpack/${PLUGIN_NAME} `;
 const VALUE_DEP_MAIN = `webpack/${PLUGIN_NAME}_hash`;
 const TYPEOF_OPERATOR_REGEXP = /^typeof\s+/;
-const WEBPACK_REQUIRE_FUNCTION_REGEXP = /__webpack_require__\s*(!?\.)/;
-const WEBPACK_REQUIRE_IDENTIFIER_REGEXP = /__webpack_require__/;
+const WEBPACK_REQUIRE_FUNCTION_REGEXP = globToRegex(
+	RuntimeGlobals.requireScope
+);
+const WEBPACK_REQUIRE_IDENTIFIER_REGEXP = globToRegex(RuntimeGlobals.require);
```

Comment:
We don't need `globToRegex` here, just use `new RegExp(...)`

Reply:
That would result in an incorrect regular expression, since the tokens would not be escaped.

```js
globToRegex(Runtimeglobals.requireScope); // /^__webpack_require__\..*$/

new RegExp(Runtimeglobals.requireScope); // /__webpack_require__.*/
```

Reply:
Just use 
```
new RegExp(`${RuntimeGlobals.requireScope}.*$`);
```

Reply:
Would you not consider that to be less generic?

Reply:
No